### PR TITLE
Initial commit: create Run_SKAI_Colab_Pipeline_Notebook.ipynb

### PR DESCRIPTION
**Pull request by ambaha1**
PR to publish the second Colab notebook material used to run the Colab pipeline notebook.

### CODE REVIEW

**Review by jzxu — CHANGES_REQUESTED** [review]

File: src/colab/Run_SKAI_Colab_Pipeline.ipynb [hunk 1]
```diff
@@ -0,0 +1,746 @@
...
+        "\n",
+        "###########################################\n",
+        "### LABELING SETTING - EMAIL PARAMETERS ###\n",
+        "###########################################\n",
+        "#@markdown ---\n",
+        "#@markdown Provide **email addresses** for all individuals that will help with labeling images, separated by commas.\n",
+        "#@markdown Emails of the labelers need to be linked to a google account.\n",
+        "\n",
+        "#@markdown ---\n",
+        "EMAIL_MANAGER = 'manager@gmail.com' #@param {type:\"string\"}\n",
```
[the remark this answers is not in the document]
Comment (ambaha1) [reply src/colab/Run_SKAI_Colab_Pipeline.ipynb]: changes on last commit [0bcee1a](https://github.com/google-research/skai/pull/72/commits/0bcee1ae52007dd5517e84cf03b728597b91c8cb)

File: src/colab/Run_SKAI_Colab_Pipeline.ipynb [hunk 2]
```diff
@@ -0,0 +1,749 @@
...
+        "  \n",
+        "  sh_command = f'bash -c \"{commands}\" | tee /tmp/output.txt'\n",
+        "  print(sh_command,'\\n')\n",
+        "\n",
+        "  if use_pexpect:\n",
+        "    return pexpect.spawn(sh_command)\n",
+        "  else:\n",
+        "    with open('/tmp/shell_command.sh', 'w') as f:\n",
+        "      f.write(commands)\n",
+        "    !bash \"/tmp/shell_command.sh\" | tee /tmp/output.txt\n",
```
Comment (jzxu) [inline src/colab/Run_SKAI_Colab_Pipeline.ipynb:135]:
```suggestion
        "    return subprocess.call('/tmp/shell_command.sh | tee /tmp/output.txt', shell=True)
```

If you make this change then you can move this function into colab_utils.py

File: src/colab/Run_SKAI_Colab_Pipeline.ipynb [hunk 3]
```diff
@@ -0,0 +1,749 @@
...
+        "\n",
+        "def load_start_tensorboard(path_log):\n",
+        "  %load_ext tensorboard\n",
+        "  %tensorboard --logdir gs://{path_log}\n",
+        "\n",
+        "colab_utils.launch_pexpect_process=launch_pexpect_process\n",
```
Comment (jzxu) [inline src/colab/Run_SKAI_Colab_Pipeline.ipynb:141]: This is a bit too tricky and error prone. Please make the change I recommended above and then move "launch_pexpect_process" into colab_utils.py.

Not sure how to fix "load_start_tensorboard". You can leave that for now and we can fix it later.

File: src/colab/Run_SKAI_Colab_Pipeline.ipynb (hunk 2 above)
Comment (ambaha1) [reply src/colab/Run_SKAI_Colab_Pipeline.ipynb]: Thanks a lot, Joseph, I think I tried using subprocess but didn't find the right way to make it works from colab_utils file
Changes pushed in commit [a590ad0](https://github.com/google-research/skai/pull/72/commits/a590ad0603eb0ccc50dbf5ec7cf409a00c910232)

**Review by jzxu — APPROVED** [review]
Thanks for addressing my comments!

File: src/colab/colab_utils.py [hunk 4]
```diff
@@ -38,6 +38,54 @@
...
+                           arguments,
+                           dir_args,
+                           use_pexpect,
+                           sleep=None):
+  """Build and run the shell command."""
+  if not isinstance(script, list):
+    script = [script]
+    arguments = [arguments]
+
+  if sleep == None and len(script) > 1:
```
Comment (jzxu) [inline src/colab/colab_utils.py:51]:
```suggestion
  if sleep is None and len(script) > 1:
```

File: src/colab/colab_utils.py [hunk 5]
```diff
@@ -38,6 +38,54 @@
...
+    sleep = [0] * (len(script) - 1)
+  elif len(script) == 1:
+    sleep == None
```
Comment (jzxu) [inline src/colab/colab_utils.py:54]:
```suggestion
    sleep = None
```

File: src/colab/colab_utils.py [hunk 6]
```diff
@@ -38,6 +38,54 @@
...
+
+  flags_str = [
+      ' '.join(f"--{f}='{v}'"
+               for f, v in argument.items())
+      for argument in arguments
+  ]
+  commands = '; '.join([
+      f'set -e', f'source {dir_args["python_env"]}',
```
Comment (jzxu) [inline src/colab/colab_utils.py:62]:
```suggestion
      'set -e', f'source {dir_args["python_env"]}',
```

**Review by jzxu — CHANGES_REQUESTED** [review]
Additional changes required by the internal lint program.

**Review by jzxu — CHANGES_REQUESTED** [review]
Can you also please take a look at the call to "launch_pexpect_process" on line 709 of colab_utils.py? It doesn't look correct and the lint program is complaining about it as well. Thanks!

**Comment by ambaha1** [comment]
Thanks for reporting on the issue on line 709 of colab_utils.py. I corrected.

**Pull request merged**